src/cpp/src/image_generation/models/unet_inference_static_bs1.hpp

```diff
@@ -15,14 +15,22 @@ namespace genai {
 // Static Batch-Size 1 variant of UNetInference
 class UNet2DConditionModel::UNetInferenceStaticBS1 : public UNet2DConditionModel::UNetInference {
 public:
+    UNetInferenceStaticBS1() : UNetInference(), m_native_batch_size(0) {}
+
+    UNetInferenceStaticBS1(const UNetInferenceStaticBS1&) = delete;
+    UNetInferenceStaticBS1(UNetInferenceStaticBS1&&) = delete;
+
+    UNetInferenceStaticBS1(UNetInferenceStaticBS1& other)
```

Comment:
We should disable direct copying for user, the `clone()` method must be the only way to create a copy. 
Let's move this constructor to the private section

Reply:
done